File: DXDAL/LotCodeSectionSettings.vb
Imports System
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpo

Namespace SPGData
    Public Class LotCodeSectionSettings
        Inherits AuditableBase

        Dim fLotCodeSection As LotCodeFormatSections
        <Association("Section-Settings")>
        Public Property LotCodeSection() As LotCodeFormatSections
            Get
                Return fLotCodeSection
            End Get
            Set(value As LotCodeFormatSections)
                SetPropertyValue("LotCodeSection", fLotCodeSection, value)
            End Set
        End Property

        Dim fSettingName As String
        <Size(50)>
        Public Property SettingName() As String
            Get
                Return fSettingName
            End Get
            Set(value As String)
                SetPropertyValue("SettingName", fSettingName, value)
            End Set
        End Property

        Dim fSettingValue As String
        <Size(100)>
        Public Property SettingValue() As String
            Get
                Return fSettingValue
            End Get
            Set(value As String)
                SetPropertyValue("SettingValue", fSettingValue, value)
            End Set
        End Property
        Public Sub New()
            MyBase.New()
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
            ' Place here your initialization code.
        End Sub

        Public Shadows Class FieldsClass
            Inherits XPLiteObject.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property Oid() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Oid"))
                End Get
            End Property
            Public ReadOnly Property LotCodeSection() As LotCodeFormatSections.FieldsClass
                Get
                    Return New LotCodeFormatSections.FieldsClass(GetNestedName("LotCodeSection"))
                End Get
            End Property
            Public ReadOnly Property SettingName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("SettingName"))
                End Get
            End Property
            Public ReadOnly Property SettingValue() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("SettingValue"))
                End Get
            End Property
        End Class
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property

    End Class
End Namespace

File: DXDAL/LotCodeSectionSettings.cs
using DevExpress.Data.Filtering;
using DevExpress.Xpo;

namespace DXDAL
{
    namespace SPGData
    {
        public class LotCodeSectionSettings : AuditableBase
        {
            private LotCodeFormatSections fLotCodeSection;
            [Association("Section-Settings")]
            public LotCodeFormatSections LotCodeSection
            {
                get => fLotCodeSection;
                set => SetPropertyValue("LotCodeSection", ref fLotCodeSection, value);
            }

            private string fSettingName;
            [Size(50)]
            public string SettingName
            {
                get => fSettingName;
                set => SetPropertyValue("SettingName", ref fSettingName, value);
            }

            private string fSettingValue;
            [Size(100)]
            public string SettingValue
            {
                get => fSettingValue;
                set => SetPropertyValue("SettingValue", ref fSettingValue, value);
            }
            public LotCodeSectionSettings() : base()
            {
            }

            public LotCodeSectionSettings(Session session) : base(session)
            {
            }

            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }

            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty Oid => new OperandProperty(GetNestedName("Oid"));
                public LotCodeFormatSections.FieldsClass LotCodeSection => new LotCodeFormatSections.FieldsClass(GetNestedName("LotCodeSection"));
                public OperandProperty SettingName => new OperandProperty(GetNestedName("SettingName"));
                public OperandProperty SettingValue => new OperandProperty(GetNestedName("SettingValue"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
